### PR TITLE
Add utility to calculate time derivative vars.

### PR DESCRIPTION
**Pull request by eslickj**
## Fixes

This adds a utility function to calculate time derivative variables after solving with PETSc. This is part of #757.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/core/solvers/petsc.py [hunk 1]
```diff
@@ -519,6 +520,30 @@ def petsc_dae_by_time_element(
...
+            if time in ComponentSet(var.get_continuousset_list()):
+                parent = var.parent_block()
+                name = var.local_name + "_disc_eq"
+                disc_eq = getattr(parent, name)
+                for i, v in var.items():
+                    try:
+                        if disc_eq[i].active:
+                            calculate_variable_from_constraint(v, disc_eq[i])
+                    except KeyError:
+                        pass # discretization equation may not exist at first time
```
Comment (andrewlee94) [inline idaes/core/solvers/petsc.py:544]: Should we check that this is the first time point in this case, and raise an error if it is not?

**Review by andrewlee94 — COMMENTED** [review]
Mostly looks good, but I have one comment on more rigorous checking of the case where the is no discretization equation.

File: idaes/core/solvers/petsc.py (hunk 1 above)
Comment (eslickj) [reply idaes/core/solvers/petsc.py:544]: That should be covered by the KeyError exception.  The first time point shouldn't have a discretization equation.

**Review by viiibhav — APPROVED** [review]
Looks good to me

File: idaes/core/solvers/petsc.py (hunk 1 above)
Comment (eslickj) [reply idaes/core/solvers/petsc.py:544]: Oh yeah, now I understand.  I'm not sure I have a clear way to know which ones should be missing with all the options.  I guess the ones that are missing should be missing.  The DAE transformation writes them so there probably not a lot of room for users to just do something weird.

**Comment by eslickj** [comment]
I put the tests in an integration tests, so codecov didn't catch them.  I added tests to unit test too.

**Review by blnicho — APPROVED** [review]

**Review by viiibhav — APPROVED** [review]
LGTM

File: idaes/core/solvers/petsc.py (hunk 1 above)
Comment (Robbybp) [reply idaes/core/solvers/petsc.py:544]: You could check this without try/except by just checking `i in disc_eq`

**Pull request merged**